src/commands/dryrun.js

```diff
@@ -0,0 +1,176 @@
+}
+
+async function extractTags(input, pid) {
+  let tags = [];
+
+  // Check if the input contains curly braces
+  if (!input.includes("{")) {
+    throw new Error("Tags not specified");
+  }
+
+  // Check if the input contains curly braces
+  if (input.includes("{")) {
+    try {
+      // Extracting content within curly braces
+      const content = input.match(/\{(.*?)\}/)[1];
+
+      const pairs = content.split(",");
+
+      // Iterating through each key-value pair
+      pairs.forEach((pair) => {
+        // Check if the pair contains an equal sign
+        if (pair.includes("=")) {
+          // Splitting each pair by "=" to get key and value
+          const [key, value] = pair.split("=").map((item) => item.trim());
+          // Normalize the value
+          let normalizedValue = value;
+          normalizedValue = normalizedValue.replace(/^['"](.*)['"]$/, "$1");
+
+          console.log("normalizedValue", normalizedValue);
+
+          // Replace "ao.id" with `pid`
+          if (normalizedValue === "ao.id") {
+            normalizedValue = pid;
+          }
+
+          //use extra quotes to use the "ao.id" word itself
+          if (normalizedValue === "'ao.id'" || normalizedValue === `ao.id"`) {
+            normalizedValue = "ao.id";
+          }
+
+          // Pushing key-value pair to tags array
+          tags.push({ name: key, value: normalizedValue });
+        } else {
+          const action = pair.trim().replace(/^'|'$/g, "");
+          tags.push({ name: "Action", value: action });
+        }
+      });
+    } catch (error) {
+      throw new Error("Invalid Syntax Usage");
+    }
+  } else {
+    // If no curly braces, assume single word input, set it as Action
+    const action = input.trim().replace(/^'|'$/g, "");
+    let normalizedAction = action;
+    normalizedAction = normalizedAction.replace(/^['"](.*)['"]$/, "$1");
+    console.log(normalizedAction);
+    tags.push({ name: "Action", value: normalizedAction });
+  }
+
+  // Adding default Target if not already present
+  let targetIncluded = tags.some((tag) => tag.name === "Target");
+  if (!targetIncluded) {
+    tags.unshift({ name: "Target", value: pid });
+  }
+
+  return tags;
+}
+
+async function callDryRun(input, pid, data, owner, id, anchor) {
+  try {
+    const tags = await extractTags(input, pid);
+    // check for Pid and data in params
+    const customProcessId = getCustomProcessId(input, pid) || pid;
+    const customData = getCustomData(input, pid) || data;
+
+    const dryrunParams = {
+      process: customProcessId,
+      tags,
+      ...(customData && { data: customData }),
+      ...(owner && { Owner: owner }),
+      ...(id && { Id: id }),
+      ...(anchor && { anchor: anchor }),
+    };
+
+    checkForVerboseFlag(input, dryrunParams);
+
+    const result = await dryrun(dryrunParams);
+    return result;
+  } catch (error) {
+    console.error(chalk.red("Error Performing DryRun:", error.message));
```

Comment:
Would it not be right to propagate the error up in context? Why swallow it here.